Missing type paths for TypeScript 4.7 Node16 modules

I believe that `rxfire` has the same TS/ESM compatibility problem as the `firebase-js-sdk` issue here: https://github.com/firebase/firebase-js-sdk/issues/6300.  The problem occurs with TS 4.7+ and ESM with the following `tsconfig.json` options:

```
{
  "compilerOptions": {
    "module": "Node16",
    "moduleResolution": "Bundler",
    "target": "ESNext",
  }
}
```

I'm actually running into the error in a project that uses `reactfire`, which depends on `rxfire`.  Compilation fails with errors like:

```
./node_modules/reactfire/dist/database.d.ts:1:29 - error TS7016: Could not find a declaration file for module 'rxfire/database'. './node_modules/rxfire/database/index.esm.js' implicitly has an 'any' type.
  There are types at './node_modules/rxfire/database/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'rxfire' library may need to update its package.json or typings.

1 import { QueryChange } from 'rxfire/database';
```

Related:
* https://github.com/firebase/firebase-js-sdk/pull/6307
* https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#esm-nodejs
* https://github.com/microsoft/TypeScript/issues/52363

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "rxfire",
-  "version": "6.0.5",
+  "version": "6.0.6",
   "private": true,
   "description": "Firebase JavaScript library RxJS",
   "author": "Firebase <firebase-support@google.com> (https://firebase.google.com/)",

